src/aead/aead.rs

```diff
@@ -77,11 +77,14 @@ impl OpeningKey {
 
 /// Authenticates and decrypts (“opens”) data in place. When
 ///
-/// The input may have a prefix that is `in_prefix_len` bytes long; any such
+/// `nonce` must be unique for every use of the key to seal data; it must be
```

Comment:
> Maybe we should advise them to ensure they always use a nonce that is unique for the given key, and to not to use random nonces since the nonce length is too small to satisfy the uniqueness requirement with random nonces.

I was going to suggest that when you wrote "I don't recommend using random nonces with these AEADs" over here https://github.com/briansmith/ring/pull/421#issuecomment-276153893, which was a total surprise to me. As it's an actual security concern and I wouldn't expect anyone except absolute crypto experts to know about this, it's something that the docs should tell you.

> In the good-but-rare cases where implicit nonces are used, the nonce will be whatever the protocol being implemented specifies it to be.

Yes. Though my proposed "the same nonce as was used for sealing" doesn't contradict that.

> In other words, the caller of `open_in_place()` doesn't have any real flexibility in choosing the nonce in any protocol.

I get your point. It's from the viewpoint of someone who uses _ring_ to implement a given protocol.

As you may have noticed, my viewpoint is usually someone who just tries to encrypt something securely and send it accross the network. If you will, such users are designing their own "protocol" (albeit a tiny one) and for their protocol design, they have to understand the fundamental concept that the nonce needs to be the same on both ends. As the Rustdoc is now, it just tells the user how to choose nonces for sealing (that is, uniquely), but for all the user knows about opening, they could again specify a new, unique and random nonce.
That's especially the case when these users come from using AES-ECB encryption, and someone told them it's insecure and they have to use AEAD instead. They don't have to change that much, but they have to know about tags and nonces now.